lib/migration.js

```diff
@@ -628,7 +628,8 @@ function mixinMigration(MySQL, mysql) {
       if (pks.length === 1) {
         const idName = this.idName(model);
         const idProp = this.getModelDefinition(model).properties[idName];
-        if (idProp.generated) {
+        const idColumnType = this.columnDataType(model, idName);
+        if (idProp.generated && idColumnType === 'INT') {
```

Comment:
I believe there are other integer type that supports auto increment in mysql, according to this docs: https://dev.mysql.com/doc/refman/8.4/en/example-auto-increment.html#:~:text=Use%20the%20smallest%20integer%20data,to%20allow%20a%20greater%20range.

Reply:
You are right @dhmlau. The other integer types also support autoincrement. I have updated the PR. Please have a look at the PR now.

Thanks for the review.